### PR TITLE
Add Junit test to AddCodeCommand, Address code review, Add PPP

### PR DESCRIPTION
**Pull request by nicholas132000**

### CODE REVIEW

File: src/test/java/seedu/clialgo/command/AddCodeCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        new AddCodeCommand("queue", "LINKED_LIST", 10).execute(topicManager, ui, fileManager, buffer);
+
+        String os = System.getProperty("os.name");
+        String expectedOutput = "";
+
+        if (os.contains("Windows")) {
+            expectedOutput = "";
+        } else {
+            expectedOutput = "";
+        }
```
Comment (heejet) [inline src/test/java/seedu/clialgo/command/AddCodeCommandTest.java:93]: Why is this blank?

**Review by heejet — CHANGES_REQUESTED** [review]

File: src/test/java/seedu/clialgo/command/AddCodeCommandTest.java (hunk 1 above)
Comment (heejet) [reply src/test/java/seedu/clialgo/command/AddCodeCommandTest.java:93]: If you expect invalid command to be printed, why is the expected output the empty string?

Comment (nicholas132000) [reply src/test/java/seedu/clialgo/command/AddCodeCommandTest.java:93]: Will make the edits!

File: src/main/java/seedu/clialgo/command/AddCodeCommand.java [hunk 2]
```diff
@@ -45,4 +47,17 @@ public void execute(TopicManager topicManager, Ui ui, FileManager fileManager, B
...
+    /**
+     * An overridden method that checks for equality of <code>AddCodeCommand</code> objects.
+     *
+     * @param otherCommand The other <code>AddCodeCommand</code> object to be checked against.
+     * @return A boolean value to determine whether the <code>AddCodeCommand</code> objects are equal.
+     */
+    @Override
+    public boolean equals(Command otherCommand) {
+        AddCodeCommand otherAddCodeCommand = (AddCodeCommand) otherCommand;
+        return Objects.equals(this.name, otherAddCodeCommand.name);
```
Comment (ong-ck) [inline src/main/java/seedu/clialgo/command/AddCodeCommand.java:60]: To check for equality of AddCodeCommand, do you also need to check whether the topic and importance are the same as well?

**Review by ong-ck — COMMENTED** [review]
Might want to replace the magic string ".cpp" as well

File: docs/team/nicholas.md
Comment (heejet) [inline docs/team/nicholas.md:1]: change the filename to ur github username.md

**Review by heejet — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/clialgo/command/AddCodeCommand.java (hunk 2 above)
Comment (nicholas132000) [reply src/main/java/seedu/clialgo/command/AddCodeCommand.java:60]: im not too sure, what do you suggest?

Comment (heejet) [reply src/main/java/seedu/clialgo/command/AddCodeCommand.java:60]: I agree with @ong-ck

Comment (ong-ck) [reply src/main/java/seedu/clialgo/command/AddCodeCommand.java:60]: Maybe you could consider this

```suggestion
        public boolean equals(Command otherCommand) {
            AddCodeCommand otherAddCodeCommand = (AddCodeCommand) otherCommand;
            boolean isNameEqual = Objects.equals(this.name, otherAddCodeCommand.name);
            boolean isTopicEqual = Objects.equals(this.topic, otherAddCodeCommand.topic);
            boolean isImportanceEqual = this.importance == otherAddCodeCommand.importance;
            return isNameEqual && isTopicEqual  && isImportanceEqual;
        }
```

Comment (nicholas132000) [reply src/main/java/seedu/clialgo/command/AddCodeCommand.java:60]: okay, will make changes

Comment (nicholas132000) [reply src/main/java/seedu/clialgo/command/AddCodeCommand.java]: is my current one okay? i combined 3 of them into the return statement

Comment (nicholas132000) [reply src/main/java/seedu/clialgo/command/AddCodeCommand.java]: ohh right

**Review by heejet — APPROVED** [review]

**Pull request merged**